gui/src/ts/components/kg/path/KgPathPage.tsx

```diff
@@ -52,18 +86,149 @@ export const KgPathPage: React.FunctionComponent = () => {
 
     return {nodes, links};
   }, [data]);
-  // console.log(pathGraphData);
+
+  // Update graph dimensions to fill flexbox container
+  // Might be bugged? For some reason, pathGraphContainerRef is always
+  // null on first render
+  React.useLayoutEffect(() => {
+    if (!pathGraphContainerRef.current) {
+      return;
+    }
+
+    // Use the height of the table and width of graph container
+    const {clientWidth} = pathGraphContainerRef.current;
+
+    if (
+      // Dimensions returned by ref can be off a little so
+      // include margin of error
+      Math.abs(pathGraphDimensions.width - clientWidth) < 10
+    ) {
+      return;
+    }
+
+    // Shave a little off the edge to avoid overflowing the flexbox container
+    // and messing up the layout
+    setPathGraphDimensions((prev) => ({...prev, width: clientWidth - 5}));
+  }, [pathGraphContainerRef.current]);
+
+  // Handle data error immediately after all hooks are run
+  if (error) {
+    return <ApolloErrorHandler error={error} />;
+  }
+
+  // Initialize color scale
+  const pathColorScale = d3.scaleOrdinal(d3.schemeCategory10);
+  data?.kgById.paths.forEach((path) => pathColorScale(path.id));
+
+  // Nested routes containing with urls and paths
+  const pathPageNestedRoutes = {
+    allPaths: {url, path},
+    path(id: string) {
+      return {
+        url: url + "/" + encodeURIComponent(id),
+        path: path + "/" + encodeURIComponent(id),
+      };
+    },
+  };
+
+  // Extract selectedPath from end of path
+  const selectedPathId = decodeURIComponent(
```

Comment:
All of this should go into a component that takes a set of paths and can be reused in different pages.

Reply:
Don't need to do that in this PR.